ldap_auth_provider.py

```diff
@@ -404,6 +400,72 @@ class _LdapConfig(object):
 
         return ldap_config
 
+    def _get_server(self, get_info=None):
+        """Constructs ServerPool from configured LDAP URIs
+
+        Args:
+            get_info ([str], optional): specifies if the server schema and server
+            specific info must be read. Defaults to None.
+
+        Returns:
+            [ServerPool]: servers groupped to ServerPool
+        """
+        return ldap3.ServerPool(
+            [
+                ldap3.Server(
+                    uri,
+                    get_info=get_info,
+                    tls=ldap3.Tls(validate=ssl.CERT_REQUIRED)
+                )
+                for uri in self.ldap_uris
+            ],
+        )
+
+    async def _fetch_root_domain(self) -> str:
+        """Fetches root domain from LDAP and saves it to ``self.ldap_root_domain``
+
+        Returns:
+            str: root domain of Active Directory forest
+        """
+        if self.ldap_root_domain is not False:
+            return self.ldap_root_domain
+
+        if self.ldap_mode != LDAPMode.SEARCH:
+            logger.warning("Fetching root domain is supported in search mode only")
```

Comment:
Is this really a warning? Is it OK to run with `ldap_active_directory` configured but not search mode? Seems like this should be an info log?

Reply:
This will arise in case some user will try to log in in form `domain\login`. This will not work in search mode.  
This warning will be printed only once and it will be usefull for debugging I think.

Reply:
I'm a bit dubious about this too. I think an `info` would be more appropriate.